services/client/src/withPagination.js

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import WayPoint from "react-waypoint";
+import Spinner from "components/spinner/loading";
+
+const withPagination = url => Component => {
+  class Pagination extends React.Component {
+    state = {
+      page: 1,
+      pageSize: 50
+    };
+
+    fetchMore = () => {
+      const {
+        props: { fetchData, hasMoreItems },
+        state: { page, pageSize }
+      } = this;
+
+      fetchData(`${url}?page=${page}&page_size=${pageSize}`).then(() => {
+        if (!hasMoreItems) return;
+        this.setState(currentState => ({ page: currentState.page + 1 }));
+      });
+    };
+
+    render() {
+      const { isLoading, hasMoreItems } = this.props;
+      return isLoading ? (
+        <Spinner />
```

Comment:
Currently, everytime data is fetched, the events component is unmounted and it shows the spinner component. To solve this, we could:
- Show the spinner inside the Events component (and every component that use pagination).
- Show the spinner as children of the Events component.

We should also add more styles to this spinner since currently, it covers all the screen.

Let me hear your thoughts on this.

Reply:
Could we render the spinner below the wrapped component? We know the user will be at the bottom of the page, so should be visible on the page

```js
<React.Fragment>
  <Component {...this.props} />
  {hasMoreItems && <WayPoint onEnter={this.fetchMore} />}
  {isLoading && <Spinner />}
</React.Fragment>
```

You may need to tweak the styles which are making it appear so large (there's some margin top and bottom set) - maybe there's a way to remove the margin on the spinner and add it to the parent for the initial state?

Reply:
Sure thing!